include/eve/detail/function/simd/riscv/compounds.hpp

```diff
@@ -0,0 +1,306 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/concept/value.hpp>
+#include <eve/detail/abi.hpp>
+#include <eve/detail/function/bit_cast.hpp>
+
+namespace eve::detail
+{
+template<plain_scalar_value T, value U, typename N>
+EVE_FORCEINLINE auto&
+self_mul(wide<T, N>& self, U const& other) noexcept
+requires std::same_as<wide<T, N>, U> && rvv_abi<abi_t<T, N>>
+{
+  constexpr auto c = categorize<wide<T, N>>();
+  if constexpr( match(c, category::integer_) ) return self = __riscv_vmul(self, other, N::value);
+  if constexpr( match(c, category::float_) ) return self = __riscv_vfmul(self, other, N::value);
+}
+template<plain_scalar_value T, value U, typename N>
+EVE_FORCEINLINE auto&
+self_mul(wide<T, N>& self, U const& other) noexcept
+requires scalar_value<U> && rvv_abi<abi_t<T, N>>
+{
+  constexpr auto c = categorize<wide<T, N>>();
+  auto           y = static_cast<T>(other);
+  if constexpr( match(c, category::integer_) ) return self = __riscv_vmul(self, y, N::value);
+  if constexpr( match(c, category::float_) ) return self = __riscv_vfmul(self, y, N::value);
+}
+template<plain_scalar_value T, value U, typename N>
```

Comment:
Please new lines between functions.

Also 

```
// self_mult --------------------

// self_add --------------------
```

This file blurs together a bit.

Reply:
Changed